sql/expression/like.go

```diff
@@ -15,6 +15,7 @@ import (
 type Like struct {
 	BinaryExpression
 	pool   *sync.Pool
+	once   sync.Once
```

Comment:
Prefer to have pointers for these sync types -- this won't work properly if we deference a Like struct pointer to make a copy of it (happens sometimes)